src/index.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  Provider as ProviderBase,
+  Fetch,
+  Executable,
+  KnownErrors,
+  TestSetupResult,
+  TestResultBundle,
+  BeaconState,
+  BeaconData,
+  ResourceTimingEntry,
+  getNetworkInformation,
+} from "@openinsights/openinsights";
+
+import { Config, ClientInfo, TaskData, Settings } from "./@types";
+
+import chooseRandomTasks from "./lib/chooseRandomTasks";
+import filterTasksByClientClassification from "./lib/filterTasksByClientClassification";
+import { normalizeEntry } from "./lib/resourceTiming";
+import { templateResource } from "./lib/templateResource";
+import { getClientInfo } from "./lib/clientInfo";
+import prefixKeys from "./util/prefixKeys";
+
+import { DEFAULT_CONFIG_URL, LIBRARY_VERSION } from "./constants";
+import retry from "./util/promiseRetry";
+
+interface ResourceCache {
+  [key: string]: string;
+}
+
+/**
+ * An OpenInsights provider for Fastly Insights.
+ */
+class FastlyProvider extends ProviderBase<Config, TaskData> {
+  private _configUrl: string;
+  private _libraryVersion: string;
+  private _resourceCache: ResourceCache = {};
+
+  /**
+   * The main constructor; it recives the settings object the site owner
+   * initialised the provider with sets required properties such as the
+   * config URL.
+   *
+   * @param settings Settings to configure the provider
```

Comment:
Do we need these parameter docs when we have typescript? I generally dispense with them, but that's cool if you're into it.

Reply:
A couple of reasons:
- Most of these param lists were copy/pasta from the openinsights provider class, which we use tsdoc to export proper documentation.
- This will allow us to export documentation for the provider if we want to in the future.